android/src/main/java/nl/xguard/flic2/Flic2.java

```diff
@@ -171,28 +172,32 @@ public void stopScan() {
 
     @Override
     public void onHostResume() {
+        Context context = getReactApplicationContext();
+        boolean isRunning = isServiceRunning(context, Flic2Service.class);
+        if (isRunning) {
+            Intent intent = new Intent(context, Flic2Service.class);
+            getReactApplicationContext().bindService(intent, mFlic2ServiceConnection, BIND_AUTO_CREATE);
+        }
     }
 
     @Override
     public void onHostPause() {
-        // empty
+        try {
```

Comment:
Aanvullende vraag:

waarom wordt er nu een unbind gedaan op de onHousePause en een bind op de Resume terwijl eerst alleen een unbind op de Destroy plaatsvond?

Is er een voordeel om de service te unbinden op hostPause? moet hij niet óók unbinden op destroy?

Reply:
Dit heeft te maken met de android lifecycle, als voorbeeld de activity lifecycle.
https://developer.android.com/guide/components/activities/activity-lifecycle

Initieel wordt de service opgestart op basis van een losse methode startService().
Om deze af service (binding) af te sluiten stond deze in de onHostDestroy() (laatste mogelijkheid om onderdelen af te sluiten/op te ruimen).
In de huidige use case, blijft de service altijd actief en wordt de methode startService() niet altijd aangeroepen.

Bij android is het niet mogelijk om te kijken of een activity-service binding, hiervoor heb ik ervoor gekozen om via onHostResume() een bind aan te roepen, als de service draait. Mocht de service niet draaien wordt er geen binding aangemaakt. Als startService() wordt aangeroepen, wordt deze alsnog opgebouwd.

Als tegenhanger van onHostResume() is onHostPause(), hierin is het dan nodig om deze op te ruimen (unbinden).
Mocht deze unbind alsnog alleen in onHostDestroy staan, dan is het mogelijk door de lifecycle dat een aantal keer onHostResume()/onHostPause() aanroept zonder onHostDestroy(), in dit scenario zou dat in theorie geen problemen opleveren aangezien Android de activity-service verbinding op de achtergrond regelt, echter is het 'netter'  om verantwoordelijkheden op hetzelfde niveau aan te maken en op te ruimen (persoonlijke smaak). Mochten we gebruiken maken van bijv een file pointer, zou dit wel problemen kunnen opleveren.


" Is er een voordeel om de service te unbinden op hostPause? moet hij niet óók unbinden op destroy?" 
Volgens de lifecycle kan enkel de onHostDestroy() aangeroepen worden, nadat onHostPause() is uitgevoerd. Hierdoor is op dit moment altijd de unbind al aanroepen.

Hopelijk is bovenstaande helder, anders stel ik voor om even telefonisch contact te hebben over dit onderwerp.

Reply:
Dankjewel 👍 we gaan over tot mergen en releasen